packages/@aws-cdk/aws-kinesisfirehose/README.md

```diff
@@ -9,8 +9,227 @@
+
+## Destinations
+
+The following destinations are supported. See [kinesisfirehose-destinations](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-kinesisfirehose-destinations-readme.html)
+for the implementations of these destinations.
+
+### S3
+
+Defining a delivery stream with an S3 bucket destination:
+
+```ts
+import * as s3 from '@aws-cdk/aws-s3';
+import * as destinations from '@aws-cdk/aws-kinesisfirehose-destinations';
+
+const bucket = new s3.Bucket(this, 'Bucket');
+
+const s3Destination = new destinations.S3Bucket(bucket);
+
+new DeliveryStream(this, 'Delivery Stream', {
+  destinations: [s3Destination],
+});
+```
+
+## Monitoring
+
+Kinesis Data Firehose is integrated with CloudWatch, so you can monitor the performance of
+your delivery streams via logs and metrics.
+
+### Logs
+
+Kinesis Data Firehose will send logs to CloudWatch when data transformation or data
+delivery fails. The CDK will enable logging by default and create a CloudWatch LogGroup
+and LogStream for your Delivery Stream.
+
+You can provide a specific log group to specify where the CDK will create the log streams
+where log events will be sent:
+
+```ts fixture=with-bucket
+import * as destinations from '@aws-cdk/aws-kinesisfirehose-destinations';
+import * as logs from '@aws-cdk/aws-logs';
+
+const logGroup = new logs.LogGroup(this, 'Log Group');
+const destination = new destinations.S3Bucket(bucket, {
+  logGroup: logGroup,
+});
+new DeliveryStream(this, 'Delivery Stream', {
+  destinations: [destination],
+});
+```
+
+Logging can also be disabled:
+
+```ts fixture=with-bucket
+import * as destinations from '@aws-cdk/aws-kinesisfirehose-destinations';
+
+const destination = new destinations.S3Bucket(bucket, {
+  logging: false,
+});
+new DeliveryStream(this, 'Delivery Stream', {
+  destinations: [destination],
+});
+```
+
+See: [Monitoring using CloudWatch Logs](https://docs.aws.amazon.com/firehose/latest/dev/monitoring-with-cloudwatch-logs.html)
+in the *Kinesis Data Firehose Developer Guide*.
+
+## Specifying an IAM role
+
+The DeliveryStream class automatically creates an IAM role with all the minimum necessary
+permissions for Kinesis Data Firehose to access the resources referenced by your delivery
+stream. For example: an Elasticsearch domain, a Redshift cluster, a backup or destination
+S3 bucket, a Lambda data transformer, an AWS Glue table schema, etc. If you wish, you may
+specify your own IAM role. It must have the correct permissions, or delivery stream
+creation or data delivery may fail.
+
+```ts fixture=with-bucket
+import * as destinations from '@aws-cdk/aws-kinesisfirehose-destinations';
+import * as iam from '@aws-cdk/aws-iam';
+
+const role = new iam.Role(this, 'Role', {
+  assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+});
+bucket.grantWrite(role);
+new DeliveryStream(this, 'Delivery Stream', {
+  destinations: [new destinations.S3Bucket(bucket)],
+  role: role,
```

Comment:
Hmm. Are you saying we won't call `bucket.grantWrite(role)` if a customer passes a Role explicitly? I feel like that's not what happens in the implementation...